src/toil_scripts/lib/programs.py

```diff
@@ -41,8 +43,24 @@ def docker_call(tool,
                                   or a url. The value is only used if mock=True
     :param dict[str,str] docker_parameters: Parameters to pass to docker
     :param bool check_output: When True, this function returns docker's output
+    :param bool return_stderr: When True, this function includes stderr in docker's output
     :param bool mock: Whether to run in mock mode. If this variable is unset, its value will be determined by
                       the environment variable.
+    
+    Piping docker command examples:
+        Running 'pipe-in-single-container' mode for command
+        'head -c 1M </dev/urandom | tee >(md5sum 1>&2) | gzip | gunzip | md5sum 1>&2':
+            command= ['head -c 1M </dev/urandom | tee >(md5sum 1>&2)', 'gzip -', 'gunzip -', 'md5sum - 1>&2']
+            work_dir=curr_work_dir
+            docker_tools=['ubuntu']
+            stdout = docker_call(work_dir=docker_work_dir, parameters=command, tool=docker_tools, check_output=True)
+        
+        Running 'pipe-of-containers' mode for command
+        'head -c 1M </dev/urandom | tee >(md5sum 1>&2) | gzip | gunzip | md5sum 1>&2':
+            command= ['head -c 1M </dev/urandom | tee >(md5sum 1>&2)', 'gzip -', 'gunzip -', 'md5sum - 1>&2']
```

Comment:
The - is necessary in order to have the piping between docker containers
work. The 4 array list is correct; one tool for each sub command in the
pipe. The main use case for it would be to run the pipe within that
container, busy just as an example it shows how the pipe between containers
works.

On Jul 28, 2016 6:38 PM, "Hannes Schmidt" <notifications@github.com> wrote:

In src/toil_scripts/lib/programs.py
<https://github.com/BD2KGenomics/toil-scripts/pull/387#discussion_r72731513>
:

>      :param bool mock: Whether to run in mock mode. If this variable is unset, its value will be determined by
>                        the environment variable.
> +
> +    Piping docker command examples:
> +        Running 'pipe-in-single-container' mode for command
> +        'head -c 1M </dev/urandom | tee >(md5sum 1>&2) | gzip | gunzip | md5sum 1>&2':
> +            command= ['head -c 1M </dev/urandom | tee >(md5sum 1>&2)', 'gzip -', 'gunzip -', 'md5sum - 1>&2']
> +            work_dir=curr_work_dir
> +            docker_tools=['ubuntu']
> +            stdout = docker_call(work_dir=docker_work_dir, parameters=command, tool=docker_tools, check_output=True)
> +
> +        Running 'pipe-of-containers' mode for command
> +        'head -c 1M </dev/urandom | tee >(md5sum 1>&2) | gzip | gunzip | md5sum 1>&2':
> +            command= ['head -c 1M </dev/urandom | tee >(md5sum 1>&2)', 'gzip -', 'gunzip -', 'md5sum - 1>&2']

The - isn't necessary for gzip and gunzip, I think. It also is inconsistent
with the example above.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/BD2KGenomics/toil-scripts/pull/387/files/bb7ab62af239974049ec09a2f835cf348411efc7#r72731513>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AJpTjFjNY3CYDKlRwgB8FqMs4Z8yWYV7ks5qaVmfgaJpZM4JVuTF>
.